content/guides/building/compiling-arm.md

```diff
@@ -16,30 +16,57 @@ if you have not already done so.
 ## Create a Compiler Toolchain
 
 Building the ARM compiler toolchain is quite easy using Haiku's ```configure``` tool.
+For a complete list of flags for the configure script, see [Haiku's Configure Options](/guides/building/configure)
 
-> For a complete list of flags for the configure script, see [Haiku's Configure Options](/guides/building/configure)
+From the Haiku source directory, run the following to compile
+the build tools (be sure to adjust the options to match your build environment):
 
-1. Perform a git clone haiku and buildtools
-2. Within the haiku source directory, create your workspace for ARM via ```mkdir generated.arm; cd generated.arm```
-2. Leverage configure to build your ARM toolchain. ```../configure -j2 --cross-tools-source ../../buildtools --build-cross-tools arm```
+```sh
+mkdir generated.arm; cd generated.arm
+../configure -j2 --cross-tools-source ../../buildtools --build-cross-tools arm
+```
 
 ## Building an MMC (SD Card) Image
 
 Once you have a complete ARM toolchain, you can build a Haiku MMC disk image via ``jam -j2 -q @minimum-mmc``
-This will generate an MMC image suitable for booting Haiku on real ARM hardware devices or in emulators like QEMU
+This will generate an MMC image suitable for booting Haiku on real ARM hardware devices or in emulators like QEMU.
 
 {{< alert-warning "Post-processing" "The generated MMC image only contains Haiku software. Most physical ARM hardware devices will require extra binary bootloaders (including u-boot). Users can leverage the Rune tool to post-process generic Haiku ARM images for their target ARM device.">}}
 
-## Emulating ARM Image
+## Building raw disk images
 
-The ARM images can also be emulated in QEMU ARM with an EFI bios like Tianocore.
+It's possible to build separate disk images for the bootloader and Haiku software. The image ``esp.image`` contains the EFI system partition with Haiku bootloader.
+The image ``haiku-minimum.image`` contains the BFS file system with Haiku kernel and software packages. These images are useful mainly for development purposes,
+when running Haiku in an emulated environment.
 
-> haiku-mmc.image contains our bootloader. haiku-minimum.mmc contains our filesystem. These need to be combined
-> in the future.  You can remove the "-hdb haiku-minimum.mmc" to see our bootloader menu on arm (or theoretically tap space?)
+```sh
+jam -j2 -q @minimum.raw esp.image haiku-minimum.image
+```
 
-The location of the ARM Tianocore bios will vary based on platform. This example is for Fedora:
+## Emulating Haiku
 
-```qemu-system-arm -M virt -m 1024 -bios /usr/share/edk2/arm/QEMU_EFI.fd -device virtio-gpu -hda haiku-mmc.image -hdb haiku-minimum.mmc```
```

Comment:
Another option is the u-boot firmware from our firmware repository.   We have a lot more control over it's functionality vs the local edk2 bios which may (or may not) exist.

https://github.com/haiku/firmware/tree/master/u-boot/arm/qemu

These firmware are also what rune uses to provision SD cards for ARM boards.

Reply:
ok that makes sense. I have an example command line below with u-boot.bin but I'll also add a link to the firmware repo.